tiledb/sm/serialization/capnp_utils.h

```diff
@@ -692,6 +694,25 @@ Status deserialize_coords(
   return Status::Ok();
 }
 
+/**
+ * Converts a JSON payload to a CAPNP message.
+ *
+ * @param buffer A span containing the message.
+ * @param message Generic CAPNP message builder.
+ * @param json Optional JsonCodec to customize the process.
+ */
+void decode_json_message(
+    span<const char> buffer,
+    auto& message_builder,
+    const ::capnp::JsonCodec& json = {}) {
+  // Trim null terminator if exists.
+  if (!buffer.empty() && buffer.back() == '\0') {
+    buffer = buffer.first(buffer.size() - 1);
+  }
+
```

Comment:
On second thought this might not be needed, considering that the null terminator gets trimmed on the server side.